libs/Microsoft.MixedReality.WebRTC/IAudioReadStream.cs

```diff
@@ -0,0 +1,22 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.MixedReality.WebRTC
+{
+    /// <summary>
+    /// High level interface for consuming WebRTC audio streams.
+    /// The implementation builds on top of the low-level AudioFrame callbacks
+    /// and handles all buffering and resampling.
+    /// </summary>
+    public interface IAudioReadStream : IDisposable
```

Comment:
@stephenatwork @djee-ms it's a bit inconsistent that we use an interface here when the rest of the public API uses classes. Using more interfaces is not necessarily a bad thing _if_ we plan to move the whole API in that direction, otherwise I'd stay consistent and just use a class with public+internal methods. Thoughts?

Reply:
I think clarity of intent trumps consistency. Thus would suggest using interfaces wherever possible, even if it's not everywhere.

Reply:
I kind of agree with @fibann, although inconsistency bothers me less than the fact that we are using an interface for hiding the implementation, without any intent of having more than 1 of them. C# is already pretty good at separating public/private API and exposing the public symbols of an assembly alone to the developers (unlike C/C++ headers), so there's no good reason to abstract that further with an interface, unless there's an intent to abstract part of the `AudioTrackReadBuffer` implementation only (if it had multiple functions; it doesn't) or expect another implementation (I don't think we do).

Reply:
> C# is already pretty good at separating public/private API and exposing the public symbols of an assembly alone to the developers

Yes, for this reason I think that using an interface as opposed to a class doesn't add much in terms of clarity of intent from the point of view of external users.

> there's no good reason to abstract that further with an interface

IMHO there are good reasons, namely you avoid dependencies on instances of a specific class in the user code, which makes refactoring easier if later you want to split it in multiple classes or merge it with another class. It's a way to write the API that has its merits - OTOH I don't think it's suited to this class more than to the rest of the existing API so I wouldn't use it just here.

Reply:
It is now a public class.